datadog_checks_dev/tests/test_docker.py

```diff
@@ -40,3 +44,60 @@ def test_compose_file(self):
             assert compose_file_active(compose_file) is False
         finally:
             run_command(['docker', 'compose', '-f', compose_file, 'down'], capture=True)
+
+    @pytest.mark.parametrize(
+        "attempts,expected_call_count",
+        [
+            [
+                None,
+                1,
+            ],
+            [
+                0,
+                1,
+            ],
+            [
+                1,
+                1,
+            ],
+            [
+                3,
+                3,
+            ],
+        ],
+    )
+    def test_retry_on_failed_conditions(self, attempts, expected_call_count):
+        compose_file = os.path.join(DOCKER_DIR, "test_default.yaml")
+
+        def condition():
+            condition.call_count += 1
+            raise RetryError("boom!")
+
+        condition.call_count = 0
+
+        try:
+            with pytest.raises(RetryError if attempts is None else tenacity.RetryError):
+                with docker_run(compose_file, attempts=attempts, conditions=[condition]):
```

Comment:
It might be easier / more appropriate to test `environment_run` instead of `docker_run` here. The main advantage is that you can do without actually starting any docker images and use dummy `up` and `down` functions instead.